### DISCUSSION TITLE
Complex processing of ite instruction under thumb instruction set

### DISCUSSION THREAD

**Discussion by imethod** [Vector35/binaryninja-api#4427] [Analysis] [answered]
Excuse me I used a translation tool

![image](https://github.com/Vector35/binaryninja-api/assets/109936536/e0a53137-5fc8-4ced-b531-46d6af596f82)

mil 
![image](https://github.com/Vector35/binaryninja-api/assets/109936536/fa77072f-39a6-47fd-b270-18517993f6cc)

The three lines of thumb instruction should be an if else structure but mil complicates it 

I tried merging variables , But the branches weren't removed

![image](https://github.com/Vector35/binaryninja-api/assets/109936536/8d0ca601-d5c5-44b8-84f8-67df155a4b6c)

ida f5  effect
![image](https://github.com/Vector35/binaryninja-api/assets/109936536/29beb0a4-dd00-4d3e-b71e-21df8dd52e0b)

ninja c
![image](https://github.com/Vector35/binaryninja-api/assets/109936536/fdb6a1b0-ba41-4d76-850d-3a6ad78f46d2)

The most important thing is that when I set the value of the variable, the second if is not recognized
![image](https://github.com/Vector35/binaryninja-api/assets/109936536/cff638b5-fa5b-4223-834e-3ffe97a376dd)

Maybe there is a function that can be optimized manually but I don't understand how 
thankes 0.0

**Comment by fuzyll** [comment, marked as answer]
Created issue #4498 out of this discussion.

**Comment by lwerdna** [comment]
@imethod Can you provide the binary used in your screenshot, or another that exhibits the same problem? Also which version of BinaryNinja are you using?

**Comment by imethod** [comment]
#4430 fix.zip 
version : Always up-to-date 3.6 dev
